vm/src/builtins/mappingproxy.rs

```diff
@@ -143,11 +143,39 @@ impl PyMappingProxy {
     fn class_getitem(cls: PyTypeRef, args: PyObjectRef, vm: &VirtualMachine) -> PyGenericAlias {
         PyGenericAlias::new(cls, args, vm)
     }
+
+    #[pymethod(magic)]
+    fn len(&self, vm: &VirtualMachine) -> PyResult<usize> {
+        let obj = self.to_object(vm)?;
+        obj.length(vm)
+    }
+
+    #[pymethod(magic)]
+    fn reversed(&self, vm: &VirtualMachine) -> PyResult {
+        let obj = self.to_object(vm)?;
+        let reversed_method = vm.get_method(obj, identifier!(vm, __reversed__)).unwrap();
+        vm.invoke(&reversed_method?, ())
```

Comment:
```suggestion
        vm.call_method(
            self.to_object(vm)?.as_object(),
            identifier!(vm, __reversed__).as_str(),
            (),
        )
```
We can simplify like this